app/models/court_date.rb

```diff
@@ -35,6 +37,14 @@ def additional_info?
   def display_name
     "#{casa_case.case_number} - Court Date - #{I18n.l(date.to_date)}"
   end
+
+  private
+
+  def set_court_report_due_date
+    if date.present? && court_report_due_date.blank?
+      self.court_report_due_date = date - 3.weeks
```

Comment:
This used to be in the controller but since we need it in both controllers I think it makes more sense to be in the model